configs/components/rubygem-puppet-resource_api.rb

```diff
@@ -0,0 +1,22 @@
+component "rubygem-puppet-resource_api" do |pkg, settings, platform|
+  pkg.load_from_json("configs/components/rubygem-puppet-resource_api.json")
+
+  pkg.build_requires "puppet-runtime"
+
+  # Install into the directory for gems shared by puppet and puppetserver
+  pkg.environment "GEM_HOME", settings[:puppet_gem_vendor_dir]
+
+  # PA-25 in order to install gems in a cross-compiled environment we need to
+  # set RUBYLIB to include puppet and hiera, so that their gemspecs can resolve
+  # hiera/version and puppet/version requires. Without this the gem install
+  # will fail by blowing out the stack.
+  pkg.environment "RUBYLIB", "#{settings[:ruby_vendordir]}:$(RUBYLIB)"
+
+  pkg.build do
+    ["gem build puppet-resource_api.gemspec"]
```

Comment:
I think this needs to use a full path. `gem` doesn't appear to be available on Solaris.

Reply:
Something like `["#{settings[:host_gem]} build puppet-resource_api.gemspec"]`

Reply:
Fixed by https://github.com/puppetlabs/puppet-agent/pull/1404.

Reply:
Thanks!